### PR TITLE
Reworked Extension Guides

### PR DESCRIPTION
**Pull request by pauljwil**
Edited and reworked content in the Extension Guides Overview and the Generator and Main Extension guides.

Signed-off-by: Paul Williams <pawilliams@mirantis.com>

### CODE REVIEW

File: docs/extensions/guides/main-extension.md [hunk 1]
```diff
@@ -73,4 +77,8 @@ export default class SamplePageMainExtension extends LensMainExtension {
 }
 ```
 
-`appMenus` is an array of objects satisfying the `MenuRegistration` interface. `MenuRegistration` extends React's `MenuItemConstructorOptions` interface. `parentId` is the id of the menu to put this menu item under (todo: is this case sensitive and how do we know what the available ids are?), `label` is the text to show on the menu item, and `click()` is called when the menu item is selected. In this example we simply log a message, but typically you would navigate to a specific page or perform some operation. Pages are associated with the [`LensRendererExtension`](renderer-extension.md) class and can be defined when you extend it. 
\ No newline at end of file
+`appMenus` is an array of objects that satisfy the `MenuRegistration` interface. `MenuRegistration` extends React's `MenuItemConstructorOptions` interface. The elements of the appMenus array are defined as follows:
```
Comment (panuhorsmalahti) [inline docs/extensions/guides/main-extension.md]:
```suggestion
`appMenus` is an array of objects that satisfy the `MenuRegistration` interface. `MenuRegistration` extends React's `MenuItemConstructorOptions` interface. The properties of the appMenus array objects are defined as follows:
```

**Review by panuhorsmalahti — DISMISSED** [review]

File: docs/extensions/guides/generator.md [hunk 2]
```diff
@@ -1,40 +1,46 @@
...
+The first thing you'll need to do is install Yeoman and Lens Extension Generator with the following command:
 
 ```bash
 npm install -g yo generator-lens-ext
 ```
 
-Run the generator and fill out a few fields for a TypeScript project:
+Run Generator by entering the following command: `yo lens-ext`.
+
+Answer the following questions in TypeScript:
```
Comment (chenhunghan) [inline docs/extensions/guides/generator.md]:
```suggestion
Answer the following questions:
```

**Review by chenhunghan — CHANGES_REQUESTED** [review]
thanks!, please remove "in TypeScript" in generator.md as I suggested in the comment.

File: docs/extensions/guides/main-extension.md [hunk 3]
```diff
@@ -20,11 +22,18 @@ export default class ExampleExtensionMain extends LensMainExtension {
...
+Two methods enable you to run custom code: `onActivate()` and `onDeactivate()`. Enabling your extension calls `onActivate()` and disabling your extension calls `onDeactivate()`. You can initiate custom code by implementing `onActivate()`. Implementing `onDeactivate()` gives you the opportunity to clean up after your extension.
 
-The following example is a little more interesting in that it accesses some Lens state data and periodically logs the name of the currently active cluster in Lens.
+Disable extensions from the Lens Extensions page:
 
-``` typescript
+1. Navigate to **File** > **Extension** in the top menu bar.
```
Comment (jim-docker) [inline docs/extensions/guides/main-extension.md]: **Extensions**

Plus of course mac has to be different, so it's **Lens** > **Extensions**

**Review by jim-docker — DISMISSED** [review]

File: docs/extensions/guides/main-extension.md (hunk 3 above)
Comment (pauljwil) [reply docs/extensions/guides/main-extension.md]: Thanks, Jim. I corrected this.

**Review by jim-docker — APPROVED** [review]

**Review by panuhorsmalahti — APPROVED** [review]

**Review by chenhunghan — APPROVED** [review]

**Pull request merged**